tools/ir-generator/irclass.cpp

```diff
@@ -251,7 +251,7 @@ void IrDefinitions::generate(std::ostream &t, std::ostream &out, std::ostream &i
 }
 
 void IrClass::generateTreeMacro(std::ostream &out) const {
-    for (auto p = this; p != nodeClass(); p = p->getParent()) out << "  ";
+    for (auto p = this; p && p != nodeClass(); p = p->getParent()) out << "  ";
```

Comment:
This is a bit odd, I am guessing it doesn't like the nullptr comparison.

Reply:
Comment was truncated. I meant to say this piece of code is a little funky in that it assumes that when p is nullptr it can terminate. The static analysis is right to add this check here but the code itself seems prone to errors.

Reply:
Yep, you are right in that, if it comes to nullptr it is a bug (elsewhere), which will now be reported.